src/liesel_gam/builder/gam_builder.py

```diff
@@ -0,0 +1,152 @@
+        self._name_counter = 0
+        self._current_predictor_prefix: str | None = None
+
+    def _get_unique_name(self, base_name: str) -> str:
+        """Generate a unique name by appending counter to base name."""
+        self._name_counter += 1
+        return f"{base_name}_{self._name_counter}"
+
+    def _get_semantic_name(self, base_name: str, unique: bool = False) -> str:
+        """Generate semantic name using predictor prefix if available."""
+        if self._current_predictor_prefix:
+            name = f"{self._current_predictor_prefix}_{base_name}"
+        else:
+            name = base_name
+
+        if unique:
+            name = self._get_unique_name(name)
+
+        return name
+
+    def terms(
+        self,
+        formula: str,
+    ) -> list:
+        """Parse formula and return list of terms.
+
+        This method does not add an default intercept term.
+
+        Args:
+            formula: R-like formula string (e.g., "x1 + x2 + s(x3, k=20)")
+
+        Returns:
+            List of Term objects from the parsed formula
+
+        Examples:
+            >>> terms = builder.terms("x1 + s(x2)")
+        """
+        return self.parser.parse_to_terms(
+            formula, default_intercept=False, name_prefix=""
+        )
+
+    def predictor(
+        self,
+        formula: str,
+        inv_link=None,
+        name: str | None = None,
+    ) -> AdditivePredictor:
+        """Create AdditivePredictor with terms from formula string.
+
+        Args:
+            formula: R-like formula string (e.g., "x1 + x2 + s(x3, k=20)")
+            inv_link: optional inverse link function (bijector)
+            name: optional custom name for the predictor
+            default_intercept: whether to include an intercept term by default
+
+        Returns:
+            AdditivePredictor with terms added according to formula
+
+        Examples:
+            >>> predictor = builder.predictor("x1 + s(x2)", name="mu")
+        """
+        predictor_name = (
+            name if name else self._get_semantic_name("predictor", unique=True)
+        )
+
+        # set predictor context for semantic naming
+        old_prefix = self._current_predictor_prefix
+        self._current_predictor_prefix = name
+
+        try:
+            # parse formula to terms using new parser system
+            terms = self.parser.parse_to_terms(
+                formula, default_intercept=True, name_prefix=name or ""
+            )
+
+            # create predictor
+            predictor = AdditivePredictor(predictor_name, inv_link=inv_link)
+
+            # add all terms to predictor
+            for term in terms:
+                predictor += term
+
+            return predictor
+        finally:
+            # restore previous prefix context
+            self._current_predictor_prefix = old_prefix
+
+    def response(
+        self, var_name: str, distribution, name: str | None = None, **params
```

Comment:
If I understand correctly, when creating a response with this method, the assumption is that each argument of the distribution is associated with a predictor in the builder, correct? What if, for one argument of the response distribution, I do not want to set up an additive predictor, but instead just a scalar variable with a certain prior?